drivers/power/max42500/max42500.c

```diff
@@ -0,0 +1,904 @@
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "max42500.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+#define CRC8_PEC        0x07      /* Implements Polynomial X^8 + X^2 + X^1 +1 */
+#define	ECOMM           9         /* General communication error */
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/******************************************************************************/
+
+/**
+ * @brief 8-bit CRC computation
+ * @param data - Data buffer.
+ * @param len - Buffer length.
+ * @return 8-bit CRC value
+ */
+static uint8_t crc8(uint8_t *data, uint8_t len)
+{
+	uint8_t crc = 0;
+	bool msb_set;
+
+	for (uint8_t idx = 0; idx < len; idx++) {
+		crc ^= data[idx];
+
+		for (uint8_t loop=0; loop < 8; loop++) {
+			msb_set = ((crc & 0x80) == 0x80);
+			crc <<= 1;
+
+			/* Divide by PEC if bit 8 was set */
+			if (msb_set)
+				crc ^= CRC8_PEC;
```

Comment:
is this PEC incompatible with [this implementation](https://github.com/analogdevicesinc/no-OS/blob/main/util/no_os_crc8.c) ?

Reply:
will update the code using that implementation